wire-ios-transport/Source/URLSession/RequestLoopDetection.swift

```diff
@@ -105,24 +102,46 @@ extension RequestLoopDetection: RequestRecorder {
 
     /// Check if there are too many occurrences of a given identifier
     private func triggerIfTooMany(identifier: IdentifierDate) {
-        if self.recordedRequests
-            .filter({ $0.identifier == identifier.identifier })
-            .count >= type(of: self).repetitionTriggerThreshold {
-            // this could be slightly faster as we don't need to count, just stop after we found N
-            // (maybe there are N+1000 entries and we don't need to go through the additional 1000)
-            self.triggerCallback(identifier.path)
-            self.recordedRequests = self.recordedRequests.filter { $0.identifier != identifier.identifier }
+        // this could be slightly faster as we don't need to count, just stop after we found N
+        // (maybe there are N+1000 entries and we don't need to go through the additional 1000)
+        let count = recordedRequests
+            .filter { $0.identifier == identifier.identifier }
+            .count
+
+        if count >= Self.repetitionTriggerThreshold {
+            triggerCallback(identifier.path)
+            recordedRequests = recordedRequests.filter { $0.identifier != identifier.identifier }
         }
     }
+
+    func isPathExcluded(_ path: String) -> Bool {
+        guard let urlComponents = URLComponents(string: path) else {
+            return false
+        }
+
+        if urlComponents.path.hasSuffix("/typing") == true {
+            return true
+        }
+
+        if urlComponents.path.hasSuffix("/search/contacts") {
+            if  let query = urlComponents.queryItems?.first(where: { $0.name == "q" }), let value = query.value {
+                return value.isEmpty
+            }
+        }
+
+        return false
+    }
 }
 
-private struct IdentifierDate {
+// MARK: -
+
+struct IdentifierDate {
     let identifier: String
     let date: Date
     let path: String
 
-    init(path: String, contentHash: UInt, date: Date) {
-        self.identifier = "\(path)[\(contentHash)]"
+    init(path: String, contentHint: String, date: Date) {
+        self.identifier = "\(path)[\(contentHint)]"
```

Comment:
I decided to switch the type to `String`, because the Objective-C hashes were `UInt` while Swift is `Int` and it's actually possible the caller adds other strings too.

Reply:
> it's actually possible the caller adds other strings too.
@markusfassbender what do you mean by that?

Reply:
In 3 places like `AssetRequestFactory` I left a comment in code:
```swift
        // [WPB-7392] through a refactoring the `contentHintForRequestLoop` was seperated from `addContentDebugInformation`.
        // Not clear if it is necessary to set `contentHintForRequestLoop` here, but keep the original behavior.
        request.addContentDebugInformation("Uploading full asset to \(path)")
        request.contentHintForRequestLoop = "Uploading full asset to \(path)"
```

Before the string's hash was used to make the request unique, now we just use the string directly instead of having some bit operations.